Added functions to inject custom Unary and Binary operators

dma82 opened the pull request:
I was in need to override the behavior of some existing operators. Now it can be done by creating an object with the operator's definition and by calling one of those new functions. Example:

```
const expr = require('expression-eval');

...

function evaluate(expression) {

  const customBinops = {
  	'>=': function(a,b) { return greaterEqual(a,b) }
  };

  const customUnops = {
	'exists': function(a) { return exists(a) }
  }

  const context = {
	search: function(content, start, end) {return search(content, start, end)}
  }
  
  expr.addCustomUnOps(customUnops);
  expr.addCustomBinOps(customBinops);
  
  return expr.compile(expression)(context);
}

```

donmccurdy commented on index.js:
```diff
@@ -205,10 +206,28 @@ function compileAsync(expression) {
-  compileAsync: compileAsync
+  compileAsync: compileAsync,
+  addCustomUnOps: addCustomUnOps,
+  addCustomBinOps: addCustomBinOps
```
I'd prefer to match the name's used by jsep (https://github.com/soney/jsep#custom-operators) for these methods:

```js
addBinaryOp('^', (a, b) => { ... });
addUnaryOp('@', (a) => { ... });
```

donmccurdy commented on index.js:
```diff
@@ -1,4 +1,5 @@
 const jsep = require('jsep');
+const _ = require('lodash');
```
Lodash would increase the size of this library quite a bit — I'd prefer not to add any new dependencies. With the API change suggested below, I don't think this PR should need it anymore.

dma82 commented:
Hi Don,

Thank you very much for taking my changes into consideration. I have changed those functions to match jsep naming convention, and removed the dependency on lodash as you suggested.

Hope it looks better now.

Thank you.

Regards

donmccurdy commented:
Do you mind having the function take just a single function at a time? So it would be used like: 

```js
import { addBinaryOp } from 'expression-eval';

addBinaryOp( '^', (a, b) => Math.pow(a, b) );
addUnaryOp( '🌞', (a) => makeSunny(a) );
```

dma82 commented:
Hi Don,

Sure there's no problem at all. I now call your library in my project in this way:

`
const expr = require('expression-eval');
const customBinaryOps = {
		':=': function(a,b) { return op.assign(a,b) },
		'>=': function(a,b) { return op.greaterEqual(a,b) },
		'<=': function(a,b) { return op.lessEqual(a,b) }
}

for (let [key, value] of Object.entries(customBinaryOps)) {
	expr.addBinaryOp(key, value);
}
`

donmccurdy commented:
Thanks!

dma82 commented:
Thanks for accepting this change and for your feedback Don! :)
